src/components/Login.tsx

```diff
@@ -1,47 +1,62 @@
 import { useRef, useState } from "react";
-import { checkValidEmail } from "./utils/validate";
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  updateProfile,
+} from "firebase/auth";
+import { Link, useNavigate } from "react-router-dom";
+import { checkValidEmail } from "../utils/validate";
 import HeaderComponent from "./ui/HeaderComponent";
-import { Link } from "react-router-dom";
-import { auth } from "./utils/firebase";
-import { createUserWithEmailAndPassword, signInWithEmailAndPassword } from "firebase/auth";
+import { auth } from "../utils/firebase";
 
 const Login: React.FC = () => {
+  type Details = {
+    email: string;
+    password: string;
+  };
   const [isSignUp, setIsSignUp] = useState<boolean>(false);
   const [errorMessage, setIsErrorMessage] = useState<string | null>(null);
-
+  const [details, setDetails] = useState<Details>({
+    email: "",
+    password: "",
+  });
   const name = useRef<HTMLInputElement>(null);
-  const email = useRef<HTMLInputElement>(null);
-  const password = useRef<HTMLInputElement>(null);
+  const navigate = useNavigate();
 
   const signUp = isSignUp ? "Already an user!" : "New to Netflix?";
   const signBtn = isSignUp ? "Sign Up" : "Sign In";
   const signUpLink = isSignUp ? "Sign in now" : "Sign up now";
-  console.log("page = ", signBtn);
+
   function handleLoginPage() {
-    setIsSignUp(prevVal => !prevVal);
+    setIsSignUp((prevVal) => !prevVal);
   }
 
-  function handleButtonClick() {
-    if (email.current && password.current) {
-      console.log("email = ", email.current.value);
-      console.log("password = ", password.current.value);
-      const message = checkValidEmail(email.current.value);
-      console.log("mesasge = ", message);
+  function handleValidate() {
+    if (details.email && details.password) {
+      const message = checkValidEmail(details.email);
       setIsErrorMessage(message);
 
       if (message) return;
 
       if (isSignUp) {
         //Signup Logic
         console.log("Entered If case");
-        createUserWithEmailAndPassword(
-          auth,
-          email.current.value,
-          password.current.value
-        )
+        createUserWithEmailAndPassword(auth, details.email, details.password)
           .then((userCredential) => {
             // Signed up
             const user = userCredential.user;
+            updateProfile(user, {
```

Comment:
Using ,then way of waiting for a action to complete can be messy in long run. To avoid nested .then hell, work with async-await and wrap the logic with a try catch to capture any error

Reply:
But this is the desired way given in firebase docs